python/dolfin/fem/form.py

```diff
@@ -1,85 +1,86 @@
 # -*- coding: utf-8 -*-
-# Copyright (C) 2017 Chris N. Richardson and Garth N. Wells
+# Copyright (C) 2017-2018 Chris N. Richardson, Garth N. Wells and Michal Habera
 #
 # This file is part of DOLFIN (https://www.fenicsproject.org)
 #
 # SPDX-License-Identifier:    LGPL-3.0-or-later
 
-"""FIXME: Add description"""
-
 import ufl
 import dolfin.cpp as cpp
 from dolfin.jit.jit import dolfin_pc, ffc_jit
 
 
-class Form(cpp.fem.Form):
-    def __init__(self, form, **kwargs):
-
-        # Check form argument
-        if not isinstance(form, ufl.Form):
-            raise RuntimeError("Expected a ufl.Form.")
-
-        sd = form.subdomain_data()
-        self.subdomains, = list(sd.values())  # Assuming single domain
-        domain, = list(sd.keys())  # Assuming single domain
-        mesh = domain.ufl_cargo()
+class Form(ufl.Form):
+    def __init__(self, form: ufl.Form, form_compiler_parameters: list=[]):
+        """Create dolfin Form
 
-        # Having a mesh in the form is a requirement
-        if mesh is None:
-            raise RuntimeError("Expecting to find a Mesh in the form.")
+        Parameters
+        ----------
+        form
+            Pure UFL form
+        form_compiler_parameters
+            Parameters used in JIT FFC compilation of this form
 
-        form_compiler_parameters = kwargs.pop("form_compiler_parameters", None)
+        Note
+        ----
+        This wrapper for UFL form is responsible for the actual FFC compilation
+        and attaching coefficients and domains specific data to the underlying
+        C++ Form.
+        """
+        self.form_compiler_parameters = form_compiler_parameters
 
         # Add DOLFIN include paths (just the Boost path for special
         # math functions is really required)
         # FIXME: move getting include paths to elsewhere
-        if form_compiler_parameters is None:
-            form_compiler_parameters = {"external_include_dirs": dolfin_pc["include_dirs"]}
+        if self.form_compiler_parameters is None:
```

Comment:
But default is now `[]` rather than `None`.

Reply:
Made default back to `None`